pandas/io/tests/test_gbq.py

```diff
@@ -721,6 +742,23 @@ def test_list_table(self):
             'Expected table list to contain table {0}'
             .format(destination_table))
 
+    def test_table_with_required_columns_shouldnt_append(self):
+        destination_table = TABLE_ID + "10"
+        test_size = 10
+        df = make_mixed_dataframe_v2(test_size)
+        schema = gbq._generate_bq_schema(df)
+        schema['fields'][0]['mode'] = 'REQUIRED'
+
+        # initialize table with same schema as the
+        # dataframe but with one REQUIRED column
+        self.table.create(destination_table, schema)
+
+        # Try appending to a table with similar schema
+        # but with REQUIRED columns, confirm failure
+        with tm.assertRaises(gbq.InvalidSchema):
+            gbq.to_gbq(df, DATASET_ID + "." + destination_table,
```

Comment:
All the datasets are numbered in the unit test environment. For example, DATASET_ID + 1 .

This line should be 
`gbq.to_gbq(df, DATASET_ID + "1." + destination_table, PROJECT_ID, chunksize=1000, if_exists='append')`